File: works/tmp/1991.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using static System.Console;

namespace ConsoleApp
{
    class Node
    {
        public char root;
        public char left;
        public char right;
        public Node(char _root, char _left, char _right)
        { root = _root; left = _left; right = _right; }
    }

    class Program
    {
        static ArrayList pre = new ArrayList();
        static ArrayList inorder = new ArrayList();
        static ArrayList post = new ArrayList();
        static Dictionary<char, Node> map = new Dictionary<char, Node>();

        static void PreOrder(char root)
        {
            if (root == '.') return;
            pre.Add(root);
            PreOrder(map[root].left);
            PreOrder(map[root].right);
        }

        static void PostOrder(char root)
        {
            if (root == '.') return;
            PostOrder(map[root].left);
            PostOrder(map[root].right);
            post.Add(root);
        }

        static void InOrder(char root)
        {
            if (root == '.') return;
            InOrder(map[root].left);
            inorder.Add(root);
            InOrder(map[root].right);
        }

        static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(Console.OpenStandardInput());
            int n = int.Parse(sr.ReadLine());
            string[] _s;
            for (int i = 0; i < n; i++)
            {
                _s = sr.ReadLine().Split(' ');
                map[_s[0][0]] = new Node(_s[0][0], _s[1][0], _s[2][0]);
            }

            PreOrder('A');
            PostOrder('A');
            InOrder('A');

            foreach (var x in pre) Write(x);
            WriteLine();
            foreach (var x in inorder) Write(x);
            WriteLine();
            foreach (var x in post) Write(x);
        }


    }

}

File: works/tmp/1991.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<char> pre, in, post;

struct tree
{
    char node;
    tree* LChild;
    tree* RChild;
};

void PreOrder(tree* root)
{
    if(root == NULL) return;
    pre.push_back(root->node);
    PreOrder(root->LChild);
    PreOrder(root->RChild);
}

void PostOrder(tree* root)
{
    if(root == NULL) return;
    PostOrder(root->LChild);
    PostOrder(root->RChild);
    post.push_back(root->node);
}

void InOrder(tree* root)
{
    if(root == NULL) return;
    InOrder(root->LChild);
    in.push_back(root->node);
    InOrder(root->RChild);
}

int main()
{
    vector<tree> v;
    for(int i = (int)'A'; i <= (int)'Z'; i++)
    {
        v.push_back({(char)i, NULL, NULL});
    }

    int n; cin >> n;
    for(int i = 0; i < n; i++)
    {
        char node, l, r;
        cin >> node >> l >> r;
        if(l != '.') v[node-'A'].LChild = &v[l-'A'];
        if(r != '.') v[node-'A'].RChild = &v[r-'A'];
    }

//    for(auto x : v)
//    {
//        cout << x.node << endl;
//        if(x.LChild != NULL) cout << x.LChild->node << endl;
//        else cout << '.' << endl;
//        if(x.RChild != NULL) cout << x.RChild->node << endl;
//        else cout << '.' << endl;
//        cout << endl;
//    }

    PreOrder(&v['A'-'A']);
    PostOrder(&v['A'-'A']);
    InOrder(&v['A'-'A']);

    for(auto x : pre) cout << x; cout << '\n';
    for(auto x : in) cout << x; cout << '\n';
    for(auto x : post) cout << x; 
}